woocommerce/woocommerce-ios
# IPP Analytics: add `card_reader_model` property to card reader software update events

Part of #5984 
Fixes #6215 


### Description


This PR aimed to add `card_reader_model` parameter to all software update events. During testing, I learned that there are two entry points to trigger software updates - from Settings > IPP during card connection and also in the connected reader screen, plus order details. However, some of the software update events are currently logged only within `CardReaderSettingsConnectedViewModel` in the connected reader screen, and that's why some software update events aren't logged from order details as in https://github.com/woocommerce/woocommerce-ios/issues/6215.

I went down the path to fix the logging issue by adding `CardReaderConnectionAnalyticsTracker` as the centralized place to log events related to card reader connection and software updates. This tracker is initialized individually in order details and IPP settings, and the software update state subscription is only active when a candidate reader is set to avoid multiple subscriptions logging the same events at the same time. Please feel free to suggest anything about this refactoring!

There are 6 software update events updated in this PR:
- `card_reader_software_update_tapped`
- `card_reader_software_update_started`
- `card_reader_software_update_failed`
- `card_reader_software_update_success`
- `card_reader_software_update_cancel_tapped`
- `card_reader_software_update_canceled`

### Testing instructions


Feel free to test as many events as possible, I tested all events on my end.

For each event, there should be a new `card_reader_model` property that shows the card reader model that is being updated. It is easier to test in a simulator with simulated card reader enabled, and `Terminal.shared.simulatorConfiguration.availableReaderUpdate = .required`.

- `card_reader_software_update_tapped`: this is triggered from IPP settings > Manage Card Reader with a connected reader and it's harder to test this case since software update is complete during the connection process. I tested this by [manually enabling "Update Reader Software" CTA](https://github.com/woocommerce/woocommerce-ios/blob/667504644115ef4b635952a20d2f17b888eeb805/WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift#L201) in the connected reader screen and tapping it
- `card_reader_software_update_started`
- `card_reader_software_update_failed`: this can be tested by turning off Wifi in the middle of the software update
- `card_reader_software_update_success`
- `card_reader_software_update_cancel_tapped`
- `card_reader_software_update_canceled`

I tested scenarios with multiple `CardConnectionController` instances around with required software update in the simulated card reader:
- Go to Settings > In-Person Payments > Manage Card Reader and wait for the card connection modal
- Cancel the software update while it's in progress --> `card_reader_software_update_started`, `card_reader_software_update_cancel_tapped`, and `card_reader_software_update_canceled` should be logged with the card reader model property
- Go to the orders tab
- Tap on an order that is eligible for IPP
- Tap "Collect Payment" CTA and wait for the software update to start
- Cancel the software update while it's in progress  --> only one set of `card_reader_software_update_started`, `card_reader_software_update_cancel_tapped`, and `card_reader_software_update_canceled` should be logged with the card reader model property

---
